q2_demux/_summarize/_visualizer.py

```diff
@@ -40,16 +40,15 @@ def __init__(self, directory_format, paired):
         self.paired = paired
 
 
-def _link_sample_n_to_file(files, counts, subsample_ns):
+def _link_sample_n_to_file(id_reads, counts, subsample_ns):
     results = collections.defaultdict(list)
     for num in subsample_ns:
         total = 0
-        for file in files:
-            sample_name = os.path.basename(file).split('_', 1)[0]
-            total += counts[sample_name]
+        for read, sample_id in id_reads:
```

Comment:
`read` doesn't really make sense, as the reads are in the file.

Let's just keep it as
`file, sample_name`
then the diff will be only this and the next line.